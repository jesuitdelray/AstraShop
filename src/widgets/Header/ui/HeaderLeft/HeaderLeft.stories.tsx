import { Provider } from "react-redux"
import { configureStore, createSlice } from "@reduxjs/toolkit"
import { Story, Meta } from "@storybook/react"
import { ModalsList } from "entities/ModalSlider"
import { HeaderLeft, HeaderLeftProps } from "./HeaderLeft"

const modalsSlice = createSlice({
    name: "modals",
    initialState: {
        current: ModalsList.NONE,
    },
    reducers: {
        openBurger: state => {
            state.current = ModalsList.BURGER
        },
        close: state => {
            state.current = ModalsList.NONE
        },
    },
})

const store = configureStore({
    reducer: {
        modals: modalsSlice.reducer,
    },
})

export default {
    title: "Widgets/HeaderLeft",
    component: HeaderLeft,
    decorators: [
        Story => (
            <Provider store={store}>
                <Story />
            </Provider>
        ),
    ],
} as Meta

const Template: Story<HeaderLeftProps> = args => <HeaderLeft {...args} />

export const Default = Template.bind({})
Default.args = {
    className: "",
    isMainPage: false,
}
