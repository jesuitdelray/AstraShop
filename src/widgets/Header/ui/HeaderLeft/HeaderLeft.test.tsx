import { render } from "@testing-library/react"
import { createStore } from "redux"
import { Provider } from "react-redux"
import { HeaderLeft } from "./HeaderLeft"

describe("HeaderLeft component", () => {
    const initialState = { currentModal: null }

    // eslint-disable-next-line default-param-last
    function reducer(state = initialState, action: any) {
        switch (action.type) {
            case "MODALS/SET_CURRENT":
                return { ...state, currentModal: action.payload }
            default:
                return state
        }
    }

    test("renders correctly with default props", () => {
        const store = createStore(reducer)

        const { container } = render(
            <Provider store={store}>
                <HeaderLeft />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    test("renders correctly with isMainPage prop", () => {
        const store = createStore(reducer)

        const { container } = render(
            <Provider store={store}>
                <HeaderLeft isMainPage={true || false} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
})
