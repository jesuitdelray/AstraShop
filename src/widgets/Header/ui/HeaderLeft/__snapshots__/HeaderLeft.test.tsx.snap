// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLeft component renders correctly with default props 1`] = `
<div>
  <div
    class="container"
  >
    <div />
  </div>
</div>
`;

exports[`HeaderLeft component renders correctly with isMainPage prop 1`] = `
<div>
  <div
    class="container"
  >
    <div />
  </div>
</div>
`;
